extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/results/ExportResultsFrameProcessor.java

```diff
@@ -107,6 +102,17 @@ public ExportResultsFrameProcessor(
       );
     }
     this.exportRowSignature = exportRowSignatureBuilder.build();
+    try {
```

Comment:
Writing to files in the constructor should be avoided since it makes the object initialization very expensive.

Reply:
Moved the connection initialization to the runIncrementally function's first run instead